ETLParser.py

```diff
@@ -0,0 +1,170 @@
+    return _table_name.match(l).group('name').lower()
+
+
+def _get_field(l):
+    '''
+    Read field information. The format is
+    'Field |Required | Data Type | Description | PEDSnet Conventions',
+    where the 'PEDSnet Conventions' field is optional.
+    '''
+    field = _field.match(l)
+
+    if not field:
+        return None
+
+    name = field.group('name').lower().strip()
+    req = field.group('req').lower().strip().startswith('yes')
+    data_type = field.group('type').strip()
+    desc = field.group('desc').strip()
+    conv = field.group('conv')
+    if conv:
+        conv = conv.strip()
+    else:
+        conv = ''
+
+    return (name, req, data_type, desc, conv)
+
+
+class Document():
+    def __init__(self, fileobj):
+        self.dict = {}
+        self.dict['content'] = ''
+        self.dict['tables'] = {}
+
+        self.fileobj = fileobj
+        self.line = ''
+
+        self.get_content()
+
+        while self.get_next_table():
+            pass
+
+    def get(self):
+        return self.dict
+
+    def add_content(self, content):
+        self.dict['content'] = content
+
+    def add_table(self, table, content=''):
+        self.dict['tables'][table] = {}
+        self.dict['tables'][table]['content'] = content
+        self.dict['tables'][table]['fields'] = {}
+
+    def add_table_content(self, table, content):
+        self.dict['tables'][table]['content'] = content
+
+    def add_field(self, table, field, required, data_type, desc, conv):
+        self.dict['tables'][table]['fields'][field] = {}
+
+        self.dict['tables'][table]['fields'][field]['required'] = required
+        self.dict['tables'][table]['fields'][field]['data_type'] = data_type
+        self.dict['tables'][table]['fields'][field]['description'] = desc
+        self.dict['tables'][table]['fields'][field]['etl_conventions'] = conv
+
+    def get_content(self):
+        # Skip everything up to and including the h1 element (the title).
+        for line in self.fileobj:
+            if _h1.match(line):
+                break
+
+        # Everything up to the first h2 element is model-level content.
+        content = []
+        for line in self.fileobj:
+            if _h2.match(line):
+                break
+            content.append(line)
+
+        self.add_content(string.join(content, '').strip())
+
+        # First h2 element is the table of contents. Skip it, up to the next h2
+        self.line = ''
+        for line in self.fileobj:
+            if _h2.match(line):
+                self.line = line
+                break
+
+    def get_next_table(self):
+        '''
+        Read from the currently cached line and up to the next h2,
+        and parse table content.
+        '''
+
+        # The document should now have the table name line cached.
+        # If it doesn't, there are no more tables to read.
+        if self.line == '':
+            return False
+
+        # Table name line looks something like this: "## 1.1 PERSON".
```

Comment:
The `_h2` regex could be updated to match on this structure `^\#\# \d+\.\d+` so other h2s would not need to be considered.